### ISSUE TITLE
SHAP values are computed incorrectly with objective "survival:aft"

### ISSUE THREAD

**Issue by Kodiologist** [dmlc/xgboost#5524]
As in #5036, the property of SHAP that the sum of SHAP values plus the bias term should equal the model predictions does not hold under the new accelerated failure time (AFT) objective, even when all observations are uncensored. This probably means that the SHAP code needs to be changed to account for AFT.

```
library(xgboost)

set.seed(5)

X <- matrix(c(1., -1.,   -1., 1.,   0., 1.,   1., 0.),
            nrow=4, ncol=2, byrow=TRUE)
dtrain <- xgb.DMatrix(X)

y <- c(2., 3., 4., 5.)
setinfo(dtrain, 'label_lower_bound', y)
setinfo(dtrain, 'label_upper_bound', y)

params <- list(objective='survival:aft',
               aft_loss_distribution='normal')
bst <- xgb.train(params, dtrain, nrounds=5)

print(cbind(
   shap.sum = rowSums(predict(bst, dtrain, predcontrib = T)),
   pred = predict(bst, dtrain)))
```

On R 3.6.2 with xgboost 88b64c8162ccfdda611590df09e0c97b9e9a400c, this prints:

```
      shap.sum     pred
[1,] 0.4576499 1.580356
[2,] 0.7656087 2.150303
[3,] 0.7656087 2.150303
[4,] 0.7656087 2.150303
```

**Comment by hcho3** [comment]
@Kodiologist I have no idea why this is happening. I need some time to read about SHAP. In the meanwhile, you can look at the full derivation of AFT loss function at https://www.overleaf.com/read/nwmzmzqktpjb. In particular, note that the tree ensemble produces outputs that mimic the log label `ln(Y)`.

**Comment by Kodiologist** [comment]
Thanks for the hint. It looks like I get agreement if I log the predictions. Should I close this as user error?

**Comment by hcho3** [comment]
@Kodiologist Ah that's interesting. This is because XGBoost applies `exp` to the output from the trees. So if the tree ensemble produces, say, -1.0, the `predict()` function will give `exp(-1.0)`. Do you think it would be better to have `predict()` function produce log predictions instead?

**Comment by Kodiologist** [comment]
The default behavior, to give predictions on the same scale as the original DV, should probably stay as it is. You might want to add a `type` argument to `predict` a la R's `predict.glm` so users can ensure they get the variable as modeled when they want to.

**Comment by hcho3** [comment]
Got it. I'll close this issue now. Let me go ahead and add a section under the AFT tutorial about using SHAP and interpreting output from `predict()`.

**Comment by davide-burba** [comment]
Do you have any hint on how to compute the shap_values in the prediction space in this case (using `shap.TreeExplainer)?
